File: c++/1305_All_Elements_in_Two_Binary_Search_Trees/solution.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int> answer;

    void dfs(TreeNode* root) {
        if (!root)
            return;
        dfs(root->left);
        answer.push_back(root->val);
        dfs(root->right);
    }
    
    vector<int> getAllElements(TreeNode* root1, TreeNode* root2) {
        dfs(root1);
        dfs(root2);
        sort(answer.begin(), answer.end());
        return answer;
    }
};

File: python/1305_All_Elements_in_Two_Binary_Search_Trees/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def getAllElements(self, root1: TreeNode, root2: TreeNode) -> List[int]:
        answer = []
        def dfs(root):
            if not root:
                return
            dfs(root.left)
            answer.append(root.val)
            dfs(root.right)
        
        dfs(root1)
        dfs(root2)
        return sorted(answer)
